envisage/tests/test_extension_registry.py

```diff
@@ -108,50 +54,20 @@ def test_remove_non_empty_extension_point(self):
         # And that the extensions are gone too.
         self.assertEqual([], registry.get_extensions("my.ep"))
 
-    def test_remove_non_existent_extension_point(self):
-        """ remove non existent extension point """
-
-        registry = self.registry
-
-        with self.assertRaises(UnknownExtensionPoint):
-            registry.remove_extension_point("my.ep")
-
-    def test_remove_non_existent_listener(self):
-        """ remove non existent listener """
-
-        registry = self.registry
-
-        def listener(registry, extension_point, added, removed, index):
-            """ Called when an extension point has changed. """
-
-            self.listener_called = (registry, extension_point, added, removed)
-
-        with self.assertRaises(ValueError):
-            registry.remove_extension_point_listener(listener)
-
     def test_set_extensions(self):
```

Comment:
Now I am also investigating into these tests, and I realized that perhaps this test should be moved to the mixin too.
The mixin is useful for testing the compliance of the IExtensionRegistry interface: `IExtensionRegistry.set_extensions` does not say subclass is allowed to raise and so the assumption is that it should not raise. `ProviderExtensionRegistry` violates the interface requirement and makes `set_extensions` always raise. So its test will also have to override the test, as a penalty for violating the interface.

Reply:
Maybe the test mixin class should be repurposed to restrict itself to unsettable registry behaviour...
Nothing to do here. Please ignore me.